pkg/apis/core/v1alpha1/storagecluster.go

```diff
@@ -107,6 +109,43 @@ type NodeSpec struct {
 	CommonConfig
 }
 
+// SecuritySpec is used to define the security configuration for a cluster.
+type SecuritySpec struct {
+	Enabled *bool     `json:"enabled,omitempty"`
+	Auth    *AuthSpec `json:"auth,omitempty"`
+}
+
+// AuthSpec lets the user define authorization configurations
+// for creating a PX Security enabled cluster
+type AuthSpec struct {
+	GuestAccess    *GuestAccessType    `json:"guestAccess,omitempty"`
+	Authenticators *AuthenticatorsSpec `json:"authenticators,omitempty"`
+}
+
+// GuestAccessType lets the user choose the
+// level of permissions the system.user has
+type GuestAccessType string
+
+var (
```

Comment:
This needs to be a var because GuestAccessType in the spec is a pointer. This is so it can be nil if the user does not provide it. When the user does not provide it, we need to set it to true by default. This is hard to do with a standard boolean